### ISSUE TITLE
ActiveRecord adapter not idempotent

### ISSUE THREAD

**Issue by wTheRockb** [jnunemaker/flipper#420]
Wondering if someone could clarify an expected behavior in the ActiveRecord adapter. Specifically in the `enable` function. If I run in my tests:
```
flipper[flag_name].enable entity
flipper[flag_name].enable entity
```
An `ActiveRecord::RecordNotUnique:` exception raises. Is this fully expected? I expect yes, since I see so many `enable` calls wrapped in `unless` blocks checking for the flag to already be enabled, but couldn't find it explicitly documented anywhere (nor addressed in tests). Additionally, in the `enable_single` method there is this line: `@gate_class.where(feature_key: feature.key, key: gate.key).destroy_all`, which I would expect to destroy any existing rows to avoid that duplicate exception.

Am I misunderstanding that line? Any help on this would be appreciated, thanks.

**Comment by xaviershay** [comment]
Not sure if related, but in https://github.com/jnunemaker/flipper/issues/422 I'm also seeing behaviour that doesn't seem to match the intent of the code.

**Comment by jnunemaker** [comment]
@wTheRockb what versions of flipper, flipper-active_record are you using?

**Comment by wTheRockb** [comment]
@jnunemaker good question! Looks like I'm a bit outdated, on `0.10.2` it looks like, which I see is outdated. I can potentially update and retest, and if I do I'll update this thread with news.

**Comment by vpereira** [comment]
I see the same issue here. I want to have a feature enabled in a trashable docker environment. Frontend and database are separated. In a flipper initializer, after the config I have the following line: 

```
Flipper[:foo].enable if Rails.env.development? 
```

My expectation was it should be idempotent. It's however not and it breaks when running the frontend docker while the flipper information already exists in the database. 

Gem version:

```
flipper (0.16.2)
flipper-active_record (0.16.2)
```

**Comment by khalilovcmd** [comment]
@wTheRockb does this issue still persists after upgrading your version?

**Comment by jnunemaker** [comment]
Closed by #436 hopefully. Use git in gemfile to check this out. I'll be releasing new version soon as well if you want to wait.
